### ISSUE TITLE
Swagger's json patch or json merge support

### ISSUE THREAD

**Issue by ahab94** [go-swagger/go-swagger#1834]
Hi, so i am new to `go-swagger`. Was testing out some of the examples. One thing that i couldn't figure is how `swagger` intends to support merge-patches and json-patches. Since patch calls are supposed to change particular attributes about an entity. What's the best way to go around with it, using swagger?

Cheers!

**Comment by fredbi** [comment]
Yes patch is problematic because empty and default Zero value are hard to distinguish, unless we use pointers (see my answer to #1830 and the original full story: #476).

You may look at previous work done in this direction with #557, but was never merged as bringing to much breaking change. At the moment and for the foreseeable future, forcing pointers using x-nullable as needed is the suggested work around.

**Comment by casualjim** [comment]
we can implement patch by providing a pre-hydrated struct but we currently haven't implemented that.

However swagger 2.0 has no provisions for json-patch at all so the answer to that question is no
